src/NServiceBus.Gateway.PersistenceTests/When_marking_message_as_dispatched.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace NServiceBus.Gateway.PersistenceTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using NUnit.Framework;
+
+    [TestFixture]
+    class When_marking_message_as_dispatched : NServiceBusGatewayDeduplicationStorageTest
+    {
+        [Test]
+        public async Task IsDuplicate_returns_true()
+        {
+            var messageId = Guid.NewGuid().ToString();
+
+            using (var session = await storage.CheckForDuplicate(messageId, new ContextBag()))
+            {
+                await session.MarkAsDispatched();
+            }
+
+            using (var session = await storage.CheckForDuplicate(messageId, new ContextBag()))
+            {
+                Assert.IsTrue(session.IsDuplicate);
+            }
+        }
+
+        [Test]
+        public async Task MarkAsDispatched_can_be_called_multiple_times()
```

Comment:
I would wrap the whole block in an Assert.DoesNotThrow. I know that if it throws the test fails as well but I think it is more expressive to express the assumptions with Asserts

Reply:
👍

Reply:
or, explicitly do `Assert.Pass();` at the end as an alternative. I personally like the test already as it is.